lms/djangoapps/instructor/views/instructor_dashboard.py

```diff
@@ -77,6 +77,17 @@ def is_enabled(cls, course, user=None):
         return bool(user and has_access(user, 'staff', course, course.id))
 
 
+def show_analytics_dashboard_message(course_key):
+    """
+    Defines whether or not the analytics dashboard URL should be displayed.
+
+    Arguments:
+        course_key (CourseLocator): The course locator to display the analytics dashboard message on.
+    """
+    return settings.ANALYTICS_DASHBOARD_URL and (not hasattr(course_key, 'ccx')
+                                                 or settings.FEATURES.get('ENABLE_CCX_ANALYTICS_DASHBOARD_URL', False))
```

Comment:
It's a little confusing to follow the logic here. Maybe separate out the CCX handling into its own section a little more? Something like:

```python
# Special Handling for CCX
if isinstance(course_key, CCXLocator):
    ccx_analytics_enabled = settings.FEATURES.get('ENABLE_CCX_ANALYTICS_DASHBOARD_URL', False)
    return settings.ANALYTICS_DASHBOARD_URL and ccx_analytics_enabled

# Normal case
return settings.ANALYTICS_DASHBOARD_URL
```

It's not as concise, but I think it's easier to read.